mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -2,69 +2,177 @@ pub mod assignment;
 
 use std::collections::HashMap;
 
-use crate::geofence::GeofenceValidator;
+use crate::{
+    geofence::{Geofence, GeofenceValidator},
+    Settings,
+};
 pub use assignment::Assignment;
 use hextree::disktree::DiskTreeMap;
 
-pub trait DiskTreeLike: Send + Sync + 'static {
-    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>>;
+pub fn make_hex_boost_data(
+    settings: &Settings,
+    usa_geofence: Geofence,
+) -> anyhow::Result<HexBoostData<UrbanizationData<DiskTreeMap, Geofence>, FootfallData<DiskTreeMap>>>
+{
+    let urban_disktree = DiskTreeMap::open(&settings.urbanization_data_set)?;
+    let footfall_disktree = DiskTreeMap::open(&settings.footfall_data_set)?;
+
+    let urbanization = UrbanizationData::new(urban_disktree, usa_geofence);
+    let footfall_data = FootfallData::new(footfall_disktree);
+    let hex_boost_data = HexBoostData::new(urbanization, footfall_data);
+
+    Ok(hex_boost_data)
 }
 
-impl DiskTreeLike for DiskTreeMap {
-    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>> {
-        self.get(cell)
-    }
+pub enum FootfallCategory {
```

Comment:
Going from a category into an assignment seems like an unnecessary extra step. Just add some comments to the assignment functions describing what's happening